Adds Airlock braces.

atlantiscze opened the pull request:
- Braces can be installed on an airlock to provide additional security. They can either be unlocked with special paired brace card (similar to an ID), or with special tool called maintenance jack (this has a decent delay).
- An airlock with active brace on it can not be opened by any means until the brace is removed. This includes AI control or anything else. Wires may be normally accessed and tampered with, but the airlock won't open even without power.
- Braces are an extra armour layer on the airlock. They absorb any damage the airlock takes until they break off completely. Braces have 600HP by default (regular airlocks have 300 for comparison). An uninstalled brace can be repaired with a welding tool.
- Braces can be obtained via fairly low-level cheap research. This includes the keycards and maintenance jacks.

It was requested by @BlueNexus

atlantiscze committed:
Adds Airlock braces.
- Braces can be installed on an airlock to provide additional security. They can either be unlocked with special paired brace card (similar to an ID), or with special tool called maintenance jack (this has a decent delay).
- An airlock with active brace on it can not be opened by any means until the brace is removed. This includes AI control or anything else. Wires may be normally accessed and tampered with, but the airlock won't open even without power.
- Braces are an extra armour layer on the airlock. They absorb any damage the airlock takes until they break off completely. Braces have 600HP by default (regular airlocks have 300 for comparison). An uninstalled brace can be repaired with a welding tool.

atlantiscze committed:
Manual changelog entry

atlantiscze committed:
Final minor code corrections.

Kasuobes commented:
Maybe allow emags to permanently break braces so they can't be repaired with a welder, by destroying the electronics inside the brace?

atlantiscze commented:
They can already be broken off with enough force (any heavy object and a bit of time will do) destroying them permanently, so i don't think special emag behavior is needed here. It's more or less a mechanical device rather than complex electronics such as an airlock itself or APC

Kasuobes commented:
Ah, so it's like a giant bar bolt. Sounds good.

BlueNexus commented:
My only comment is that the brace itself should be more expensive materials-wise, based on the size of the brace.

BlueNexus commented on code/modules/research/designs.dm:
```diff
@@ -1709,3 +1709,32 @@ CIRCUITS BELOW
+	name = "airlock brace design"
+	desc = "Special door attachment that can be used to provide extra security."
+	id = "brace"
+	req_tech = list(TECH_ENGINEERING = 3, TECH_POWER = 2)
```
Wouldn't materials tech be more appropriate than power tech, as the brace doesn't use power?

BlueNexus commented on code/modules/research/designs.dm:
```diff
@@ -1709,3 +1709,32 @@ CIRCUITS BELOW
+	name = "maintenance jack design"
+	desc = "A special maintenance tool that can be used to remove airlock braces."
+	id = "bracejack"
+	req_tech = list(TECH_ENGINEERING = 3, TECH_POWER = 2)
```
Same as the above comment.

atlantiscze committed:
Increases price of airlock brace and tweaks research level requirements

atlantiscze commented:
Material cost has been increased, now costs about two steel sheets. Given it absorbs 600HP of damage it sounds apropriate. TECH_POWER requirement replaced with TECH_MATERIAL.

BlueNexus commented:
:ok_hand:

GinjaNinja32 commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -536,7 +537,10 @@ About the new airlock wires panel:
+
+	if(brace)
+		brace.update_icon()
+		overlays += image(brace.icon, brace.icon_state)
```
`overlays += brace`, you don't need to make an image for it :P

PsiOmegaDelta replied:
It is more memory efficient to make images rather than use strings, if it's indeed true that using strings includes the entire dmi in the resulting icon while using images does not.

GinjaNinja32 replied:
This isn't a string, though, it's an obj :P

atlantiscze replied:
So, should i replace this with overlays += brace or not?

mwerezak replied:
It's fine the way it is IMO

GinjaNinja32 replied:
depends if braces will ever need overlays/underlays/colors/etc. if they get any of those, this won't work.

mwerezak commented:
Any screenshots?

Raptor1628 commented:
Who has maintenance jacks?

Hubblenaut commented:
Normal tools will do absolutely nothing?

LorenLuke commented:
Is there any way around it that can be accomplished via subtlety or intelligence, compare breaking down a door versus picking it's lock?

atlantiscze commented:
@mwerezak: Here's one screenshot:
![image](https://cloud.githubusercontent.com/assets/5147960/16356869/ab46a172-3ae6-11e6-990b-33a44bf58955.png)
On the left, an airlock with brace applied. On the right, from top down: keycard, maintenance jack, brace itself (undeployed state)

@Raptor1628: Currently, maintenance jacks, braces and keycards are research-printed only. I may map them in in separate PR later, due to tendency of PRs with map changes to break down recently.

@Hubblenaut: You can still hack the airlock as usual, but you can't force it open with crowbar. As for harm intent, any tool that would be able to damage the airlock physically will be able to damage the brace.

@LorenLuke: Three ways of removing brace exist: Brute force, Maintenance jack (takes 15-30s) and (paired) keycard (takes 1s).

Raptor1628 commented:
Might want to make the maintenance jack have some advantages and uses other than only being for these. Or add the functionality of working as a maintenance jack to other tools. Otherwise, if these become common, carrying around a tool just to open them that does nothing else is going to be a huge waste of space.

atlantiscze commented:
Maintenance jack already has an advantage - it is able to open any airlock brace, while keycards must be paired with the brace first.

Raptor1628 commented:
I mean more, most, if not all tools have multiple uses. The maintenance jack will be a one trick pony, probably decently sized, only for the purpose of opening maintenance braces. At the very least let it act like a crowbar.

BlueNexus commented:
@Raptor1628 To be fair, it isn't crowbar-shaped at all. It has a hammer and whatever the thing on the end is.

Raptor1628 commented:
The back end of the hammer looks like it could be used for prying, and the end could be for a wrench too. I mean including this entirely as a reference to Alien: Isolation is one thing, but we should make it so that it's at least integrated into the game a bit better.

mwerezak commented:
I wouldn't mind if it was also a crowbar subtype. Crowbars are often used for manipulating doors, it seems like a pairing that makes sense.

GinjaNinja32 commented:
I don't like adding a snowflake tool that has no use other than to remove airlock braces and that is the only thing that can do that. Perhaps have it be the /best/ thing to remove a brace with, but allow a lengthy sequence with a standard toolbelt (something with crowbar/wrench/screwdriver a few times, like r-walls?), and allow the jack to be used for other things than just braces (as Raptor and Harpy said, crowbar would make sense, as would wrench).

atlantiscze committed:
Feedback tweaks
- Brace's health reduced by 25% (600 -> 450)
- Maintenance jack is now a wrench subtype, which means it should be able to do pretty much anything a wrench can do, with added advantage of being able to remove airlock braces.

atlantiscze commented:
Maintenance jacks can now be used as a wrench alternative.
@Mwerezak: I've reduced the brace's health to 450. I think it should be more resistant than the airlock itself, even if by a small amount.

LorenLuke commented:
I have a question: What with all manner of things and reasons to open doors that prevent themselves being opened, i.e. rogue AI, someone dying in a department, gaining access to an unstaffed department with no synths, etc... what reason exists for keeping a door shut that supercedes anything above bolting, access locking, or depowering it? I can understand the idea of a scissor-jack type thing that eventually forces doors open, but a brace to keep them from never opening (within the standard realm of corporate operation), I don't understand.

BlueNexus commented:
@LorenLuke the main purpose I intended was to give people a way to securely hold closed depowered doors. The only alternative is welding, which can be undone instantly.

LorenLuke commented:
Perhaps you just have it grant door power while attached, and have it have its own power supply? Additional options include interfacing with the ID card reader and stuff.

BlueNexus commented:
@LorenLuke That would make it useless if the door it's applied to is powered. Its purpose is to hold doors closed, not make it possible to open them. Plus, the jack is a purely mechanical device, it would have no way to transfer power anyway. I don't see any issues with the PR's current implementation.

mwerezak commented:
> @LorenLuke the main purpose I intended was to give people a way to securely hold closed depowered doors. The only alternative is welding, which can be undone instantly with a freely accessible tool.

Um, door bolts exist you know. Not to say that having some alternative isn't welcome.

mwerezak commented:
> @Mwerezak: I've reduced the brace's health to 450. I think it should be more resistant than the airlock itself, even if by a small amount.

I'm not really seeing why it would need to have more health than an airlock, but okay, I guess 450 is a decent starting point.

BlueNexus commented:
> Door bolts exist you know. Not to say that having some alternative isn't welcome.

Yes, but door bolts cannot be raised again if the door is unpowered. Braces are reversible.

LorenLuke commented:
Can't that be accomplished by a self contained device that powers the door?

mwerezak commented on code/game/machinery/doors/braces.dm:
```diff
@@ -11,9 +11,9 @@
 
 // MAINTENANCE JACK - Acts as an universal keycard, but works with a 15-30s delay
-/obj/item/weapon/brace_jack
+/obj/item/weapon/wrench/brace_jack
```
I'd prefer if it could also be used as a crowbar rather than a wrench. Crowbars are often used with doors and I don't see any way for wrenches to interact with door braces so making it a crowbar subtype would make more sense IMO.

atlantiscze committed:
Maintenance jack is now a crowbar subtype

BlueNexus commented on code/game/machinery/doors/braces.dm:
```diff
@@ -95,13 +95,13 @@
 		user << "You begin forcibly removing \the [src] with \the [C]."
 		if(do_after(user, rand(150,300), airlock))
-			user << "You finish removing \the [src]."
+			user << "You finissh removing \the [src]."
```
Finissh?

mwerezak commented:
A brace health of 300 still sounds better to me, but :+1: either way.

PsiOmegaDelta commented:
Awaiting typo fix.

Hubblenaut commented:
Maintenance Jacks look so big, could they be slung over the back with a nice extra back slot sprite? I imagine them to be on par with a fire axe.

PsiOmegaDelta commented:
Can be arranged separately.
